common/include/nephtys/utils/config.hpp

```diff
@@ -0,0 +1,66 @@
+//
+// Created by sztergbaum roman on 2019-04-19.
+//
+
+#pragma once
+
+#include <filesystem>
+#include <fstream>
+#include <nlohmann/json.hpp>
+#include <loguru.hpp>
+
+namespace nephtys::utils
+{
+    namespace details
+    {
+        template<typename TConfig>
+        TConfig create_configuration(std::filesystem::path &&config_path,
```

Comment:
I don't think you need to force a rvalue reference here, since you never need ownership over the path, you just use it to create directories